src/map/battlefield_handler.cpp

```diff
@@ -36,183 +36,122 @@
 #include "recast_container.h"
 #include "status_effect_container.h"
 
-CBattlefieldHandler::CBattlefieldHandler(uint16 zoneid)
-{
+CBattlefieldHandler::CBattlefieldHandler(uint16 zoneid) {
     m_ZoneId = zoneid;
 
-    //Dynamis zone (need to add COP dyna zone)
-    //added ghelsba outpost here, 1 battlefield only
-    if (m_ZoneId > 184 && m_ZoneId < 189 || m_ZoneId > 133 && m_ZoneId < 136 || m_ZoneId == 140 || m_ZoneId == 35 || m_ZoneId > 38 && m_ZoneId < 43)
-    {
+    // The Garden of RuHmet, Dynamis zones, and Ghelsba Outpost
+    if (m_ZoneId == 35 || m_ZoneId > 38 && m_ZoneId < 43 || m_ZoneId > 133 && m_ZoneId < 136 || m_ZoneId == 140 || m_ZoneId > 184 && m_ZoneId < 189 ) {
         m_MaxBattlefields = 1;
+
+    // Temenos
+    } else if (m_ZoneId == 37) {
+        m_MaxBattlefields = 8;
+
+    // Apollyon
+    } else if (m_ZoneId == 38) {
+        m_MaxBattlefields = 6;
+
+    // All other battlefield zones
+    } else {
+        m_MaxBattlefields = 3;
     }
-    else
-        if (m_ZoneId == 37)
-        {
-            m_MaxBattlefields = 8;
-        }
-        else
-            if (m_ZoneId == 38)
-            {
-                m_MaxBattlefields = 6;
-            }
-            else
-            {
-                m_MaxBattlefields = 3;
-            }
+
     memset(&m_Battlefields, 0, sizeof(m_Battlefields));
 }
 
+/***********************************************************
+ is battlefield bnum free
+ ***********************************************************/
+bool CBattlefieldHandler::isBattlefieldFree(uint16 bnum) {
+    if (m_Battlefields[bnum-1] == nullptr)
+        return true;
+    else
+        return false;
+}
+
 void CBattlefieldHandler::handleBattlefields(time_point tick) {
     for (int i = 0; i < m_MaxBattlefields; i++) {
         if (m_Battlefields[i] != nullptr) { //handle it!
             CBattlefield* PBattlefield = m_Battlefields[i];
             int instzone = PBattlefield->getZoneId();
 
-            //handle time remaining prompts (since its useful!) Prompts every minute
+            // update timers
             auto time_elapsed = std::chrono::duration_cast<std::chrono::seconds>(tick - PBattlefield->getStartTime());
```

Comment:
maybe move this to lua like instances do or on my branch
https://github.com/DarkstarProject/darkstar/blob/bcnmshit/scripts/globals/battlefield.lua#L57

also i think kj was saying the way instances do it is how retail handles it (can grab https://github.com/DarkstarProject/darkstar/blob/bcnmshit/scripts/globals/battlefield.lua#L110 if you feel like handling it in lua instead)